istioctl/cmd/dashboard.go

```diff
@@ -510,17 +525,35 @@ func dashboard() *cobra.Command {
 	dashboardCmd.PersistentFlags().StringVarP(&addonNamespace, "namespace", "n", istioNamespace,
 		"Namespace where the addon is running, if not specified, istio-system would be used")
 
-	dashboardCmd.AddCommand(kialiDashCmd())
-	dashboardCmd.AddCommand(promDashCmd())
-	dashboardCmd.AddCommand(grafanaDashCmd())
-	dashboardCmd.AddCommand(jaegerDashCmd())
-	dashboardCmd.AddCommand(zipkinDashCmd())
-	dashboardCmd.AddCommand(skywalkingDashCmd())
+	kiali := kialiDashCmd()
+	kiali.PersistentFlags().IntVar(&kialiPort, "kiali_port", defaultKialiPort, "Kiali port")
+	dashboardCmd.AddCommand(kiali)
+
+	prom := promDashCmd()
+	prom.PersistentFlags().IntVar(&promPort, "prom_port", defaultPrometheusPort, "Prometheus port")
+	dashboardCmd.AddCommand(prom)
+
+	graf := grafanaDashCmd()
+	graf.PersistentFlags().IntVar(&grafanaPort, "grafana_port", defaultGrafanaPort, "Grafana port")
```

Comment:
```suggestion
	graf.PersistentFlags().IntVar(&grafanaPort, "grafana-port", defaultGrafanaPort, "Grafana port")
```

I would just have a single `--port` one though, we already know if its "grafana" or "kiali" based on the dashboard command

Reply:
`--port` has already been used for the local port to listen to. I just uniformed to a `--ui-port` flag for each command.